src/hooks/useSpawnEngine.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+
+import { spawnEngine } from '@/engine/spawn'
+
+import { useOnce } from './useOnce'
+
+export function useSpawnEngine() {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const spawnAsync = () => {
+    const spawn = async () => {
+      setIsLoading(true)
+
+      try {
+        await spawnEngine()
+      } catch (error) {
+        if (!isWeb()) throw error
+        console.warn(
+          `Running in the browser: cannot spawn sidecar with @tauri-apps/api/shell. Make sure engine is running:
+                    npm run sidecar:exec`
+        )
+      }
+
+      setIsLoading(false)
+    }
+    spawn()
+  }
+
+  const once = useOnce()
+
+  useEffect(() => {
+    once.do(spawnAsync)
+  }, [once])
```

Comment:
Nice extraction. What do you think of a similar usage as `useEffect` for `useOnce`? Like:
```tsx
  useEffectOnce(() => {
    spawn()
  })
```

Reply:
I like its composability a lot as it is. I will keep it with the current api for this pr and we may refactor later.